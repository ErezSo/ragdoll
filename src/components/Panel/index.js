import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 20px;
`;
const Active = styled.span`
  color: #1cc51c;
  margin: 0 10px;
`;
const Inactive = styled.span`
  color: red;
  margin: 0 10px;
`;

const Panel = ({ devices, handleSearch }) => {
  const numOfActiveDevices = devices.filter(d => d.active).length;
  return (
    <Wrapper>
      <div>
        Active devices:
        <Active>{numOfActiveDevices}</Active>
        {"/"}
        <Inactive>{devices.length - numOfActiveDevices}</Inactive>
      </div>
      <div>
        Search: <input type="text" onChange={handleSearch} />
      </div>
    </Wrapper>
  );
};

export default Panel;
