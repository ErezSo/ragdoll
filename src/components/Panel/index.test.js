import React from "react";
import { shallow } from "enzyme";
import Panel, { Active, Inactive } from "./";

describe("Panel", () => {
  let wrapper;
  const devices = [
    {
      name: "rotation_rate_gamma",
      unit: "deg/s",
      value: 0,
      timestamp: 1528878066506,
      active: true
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<Panel devices={devices} />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("render the number of active an inactive devices", () => {
    expect(
      wrapper
        .find(Active)
        .dive()
        .text()
    ).toBe("1");

    expect(
      wrapper
        .find(Inactive)
        .dive()
        .text()
    ).toBe("0");
  });
});
