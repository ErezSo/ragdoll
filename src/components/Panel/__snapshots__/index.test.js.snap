// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly 1`] = `
"<styled.div>
  <div>
    Active devices:
    <styled.span>
      1
    </styled.span>
    /
    <styled.span>
      0
    </styled.span>
  </div>
  <div>
    Search: 
    <input type=\\"text\\" onChange={[undefined]} />
  </div>
</styled.div>"
`;
